freetextresponse/tests.py

```diff
@@ -35,61 +61,165 @@ def make_an_xblock(cls, **kw):
+        result = ValidationMessage(
+            ValidationMessage.ERROR,
+            _(msg)
         )
-        self.assertIn(
-            ', '.join(
-                self.xblock.halfcredit_keyphrases,
-            ),
-            studio_view_html,
+        test_result = FreeTextResponse._generate_validation_message(msg)
+        self.assertEquals(
+            type(result),
+            type(test_result),
+        )
+        self.assertEquals(
+            result.text,
+            test_result.text,
+        )
+
+    @ddt.file_data('./tests/validate_field_data.json')
+    def test_validate_field_data(self, **test_dict):
+        """
+        Checks classmethod validate_field_data
+        tests the instuctor values set in edit
+        """
+        print test_dict
+        test_data = TestData()
+        test_data.weight = test_dict['weight']
+        test_data.max_attempts = test_dict['max_attempts']
+        test_data.max_word_count = test_dict['max_word_count']
+        test_data.min_word_count = test_dict['min_word_count']
+        test_data.submitted_message = test_dict['submitted_message']
+        validation = set()
+        self.xblock.validate_field_data(validation, test_data)
+        validation_list = list(validation)
+        # Only one validation error should be in set
+        self.assertTrue(len(validation_list) == 1)
+        self.assertEquals(
+            test_dict['result'],
+            validation_list[0].text,
+        )
+
+    def test_get_resource_string(self):
+        # pylint: disable=protected-access
+        """
+        Checks that get_resource_string returns the proper html
+        """
+        student_view_html = self.xblock.student_view().content
+        test_result = FreeTextResponse.get_resource_string('view.html')
+        test_result = test_result.format(
+            self=self.xblock,
+            word_count_message=self.xblock._get_word_count_message(),
+            indicator_class=self.xblock._get_indicator_class(),
+            problem_progress=self.xblock._get_problem_progress(),
+            used_attempts_feedback=self.xblock._get_used_attempts_feedback(),
+            nodisplay_class=self.xblock._get_nodisplay_class(),
+            visibility_class=self.xblock._get_indicator_visibility_class(),
+            submitted_message='',
+            user_alert='',
+        )
+        self.assertEquals(student_view_html, test_result)
+
+    @ddt.data('view.js.min.js', 'view.less.min.css')
+    def test_get_resource_url(self, path):
+        """
+        Checks that get_resource_url the correct url
+        """
+        public_path = '/resource/freetextresponse/public'
+        result = "{public_path}/{path}".format(
+            public_path=public_path,
+            path=path,
+        )
+        self.xblock.runtime.local_resource_url = MagicMock(
+            return_value=result
+        )
+        test_result = self.xblock.get_resource_url(path)
+        self.assertTrue(result == test_result)
```

Comment:
Let's avoid this idiom anywhere it occurs.
If we're checking equality, always use `assertEquals`.